source/row_selector.ts

```diff
@@ -0,0 +1,316 @@
+import { ArrayTableModel } from './array_table_model';
+import { TableModel } from './table_model';
+import { AddRowOperation, MoveRowOperation, Operation }
+  from './operations';
+
+/** Provides the functionality needed to resize a table's columns. */
+export class RowSelector {
+
+  /** Constructs a ColumnResizer.
+   * @param table - The interface to the table being resized.
+   */
+  constructor(table: TableModel) {
+    this.isShiftDown = false;
```

Comment:
Initialize variables in the order of their declaration, either reorder their declaration or reorder their initialization.